tests/flow/test_bruteforce.py

```diff
@@ -3,6 +3,79 @@
 # the Server Side Public License v1 (SSPLv1).
 
 from common import *
+def test_bytearray():
+    dim = 16
+    num_elements = 10
+    k=10
+
+    params = VecSimParams()
+    bfparams = BFParams()
+
+    params.algo = VecSimAlgo_BF
+    bfparams.initialCapacity = num_elements
+    bfparams.blockSize = num_elements
+    bfparams.dim = dim
+    bfparams.type = VecSimType_FLOAT32
+    bfparams.metric = VecSimMetric_Cosine
+
+    params.bfParams = bfparams
+
+    bfindices = []
+
+    fp32_cosine_index = VecSimIndex(params)
+    bfindices.append(fp32_cosine_index)
+
+    params.bfParams.metric = VecSimMetric_L2
+    fp32_L2_index = VecSimIndex(params)
+    bfindices.append(fp32_L2_index)
+
+    params.bfParams.type = VecSimType_FLOAT64
+    fp64_L2_index = VecSimIndex(params)
+    bfindices.append(fp64_L2_index)
+
+    params.bfParams.metric = VecSimMetric_Cosine
+    fp64_cosine_index = VecSimIndex(params)
+    bfindices.append(fp64_cosine_index)
+
+    np.random.seed(47)
+    fp32_data = np.float32(np.random.random((num_elements, dim)))
+    fp32_vectors = []
+    fp64_data = np.float64(np.random.random((num_elements, dim)))
+    fp64_vectors = []
+
+    for i, vector in enumerate(fp32_data):
+        fp32_vectors.append((i, vector))
+        for index in fp32_cosine_index, fp32_L2_index:
+            index.add_vector(vector, i)
+
+    for i, vector in enumerate(fp64_data):
+        fp64_vectors.append((i, vector))
+        for index in fp64_cosine_index, fp64_L2_index:
+            index.add_vector(vector, i)
+
+    fp32_query_data = np.float32(np.random.random((1, dim)))
+    fp64_query_data = np.float64(np.random.random((1, dim)))
+
+
+    for(query, vectors, index) in (fp32_query_data, fp32_vectors, fp32_cosine_index), (fp64_query_data, fp64_vectors, fp64_cosine_index):
```

Comment:
This loop looks like it will be hard to maintain. Consider zipping the query, data and indexes (and the python distance function) in into an object and then iterating on it for testing each index